### ISSUE TITLE
Incorrect detection of copy-constructor

### ISSUE THREAD

**Issue by abbyssoul** [L2Program/FlintPlusPlus#53]
This code example produces incorrect diagnostics on the master version of flint+++

> cat flint_bug.cpp

``` C++
struct C {};
struct B {};

struct CC {
    CC(C& a, B& b) {}
};
```

> flint++ -v -r flint_bug.cpp 
> [Error  ] flint_bug.cpp:7: Copy constructors should take a const argument: CC(C& a, B& b)
> 
> Lint Summary: 1 files
> Errors: 1 Warnings: 0 Advice: 0

Please note that "CC(C& a, B& b)"  is not a copy constructor. If I change the name of the struct C -> E everything works fine. So my guess is the name of the first argument 'collaids' with the name of the class and thus the constructor is identified as a copy constructor. This happened on a large code base where class names are not single letters. So for example the following also trip the parser:

``` C++
struct SomeType {};
struct B {};

struct SomeTypeButDifferent {
    SomeTypeButDifferent(SomeType& a, B& b) {}
};
```

**Comment by alarobric** [comment]
Reproduced this myself as well.
